src/D2L.CodeStyle.Analyzers/Immutability/KnownImmutableTypes.cs

```diff
@@ -74,19 +72,7 @@ internal sealed class KnownImmutableTypes {
 			SpecialType.System_UIntPtr
 		);
 
-		/// <summary>
-		/// A list of known immutable types defined for the assembly.
-		/// </summary>
-		private readonly ImmutableHashSet<string> DeclaredKnownImmutableTypes;
-
-		internal KnownImmutableTypes( ImmutableHashSet<string> declaredKnownImmutableTypes ) {
-			DeclaredKnownImmutableTypes = declaredKnownImmutableTypes;
-		}
-
-		internal KnownImmutableTypes( IAssemblySymbol a )
-			: this( LoadFromAssembly( a ).ToImmutableHashSet() ) { }
-
-		internal bool IsTypeKnownImmutable( ITypeSymbol type ) {
+		internal static bool IsTypeKnownImmutable( ITypeSymbol type ) {

```

Comment:
This may not stay `static` because I want to remove the stringly-based comparison of symbols in here but for now this should do.